modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointWorkflow.java

```diff
@@ -214,7 +217,9 @@ public void markCheckpointEnd(Checkpoint chp) throws IgniteInternalCheckedExcept
             }
         }
 
-        checkpointMarkersStorage.onCheckpointEnd(chp.progress.id());
+        if (chp.hasDelta()) {
```

Comment:
Why is this required?

Reply:
If there are no dirty pages, then checkpoint markers are not needed.

Reply:
Ok, I see now